code/iaas/logic/src/main/java/io/cattle/platform/process/instance/InstancePreStart.java

```diff
@@ -66,7 +66,12 @@ public void doWithLockNoResult() {
             Map<String, Object> driver_opts = DataAccessor.fieldMap(v, VolumeConstants.FIELD_VOLUME_DRIVER_OPTS);
             if (driver_opts.containsKey(VolumeConstants.EC2_AZ)) {
                 String zone = driver_opts.get(VolumeConstants.EC2_AZ).toString();
-                labels.put(HostAffinityConstraint.LABEL_HEADER_AFFINITY_HOST_LABEL, String.format("%s=%s", VolumeConstants.HOST_ZONE_LABEL_KEY, zone));
+                String label = String.format("%s=%s", VolumeConstants.HOST_ZONE_LABEL_KEY, zone);
+                if (labels.containsKey(HostAffinityConstraint.LABEL_HEADER_AFFINITY_HOST_LABEL)) {
```

Comment:
I don't see any need for two separate checks. Just one:
```
if (!StringUtils.isEmpty(labels.containsKey(HostAffinityConstraint.LABEL_HEADER_AFFINITY_HOST_LABEL))) {
    label = labels.get(HostAffinityConstraint.LABEL_HEADER_AFFINITY_HOST_LABEL).toString() + "," + label;
}
```
Am I missing something?
Also, `!isEmpty` is better than `!= null` because it will prevent the weird scenario where that label is set with a empty string and you get a value like `",,io.rancher.host.zone=east"`. I realize that this seems overkill but its good defensive programming with no extra cost in complexity.